pennylane/resource/resource.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The data class which will aggregate all the resource information from a quantum workflow.
+"""
+from dataclasses import dataclass, field
+
+
+@dataclass(frozen=True)
+class Resources:
+    r"""Contains attributes which store key resources such as number of gates, number of wires, shots,
+    depth and gate types.
+
+    Args:
+        num_wires (int): number of qubits
+        num_gates (int): number of gates
+        gate_types (dict): dictionary storing operation names (str) as keys
+            and the number of times they are used in the circuit (int) as values
+        depth (int): the depth of the circuit defined as the maximum number of non-parallel operations
+        shots (int): number of samples to generate
+
+    Raises:
+        TypeError: If the attributes provided are not of the correct type.
+        ValueError: If an attribute provided has value less than 0.
+        AttributeError: If an attribute is set after initialization.
```

Comment:
Are these still the case?

Reply:
Nice catch! They are not, removing now !